js/co-design.js

```diff
@@ -0,0 +1,275 @@
+                    this.refersToElement.style.right = (document.body.clientWidth - rect.right) + "px";
+                    offset = document.body.clientWidth - rect.right;
+                }
+                else {
+                    this.refersToElement.style.left = (rect.left) + "px";
+                    offset = rect.left;
+                }
+                if (document.body.clientWidth < (refersToElementWidth + offset)) {
+                    this.refersToElement.style.left = "0px";
+                    this.refersToElement.style.right = "0px";
+                    this.refersToElement.style.maxWidth = "100%";
+                }
+            };
+            HeaderSlideToggle.find = function (container, query) {
+                var elements = container.querySelectorAll(query);
+                var toggles = [];
+                for (var i = 0; i < elements.length; i++) {
+                    toggles.push(new HeaderSlideToggle(elements[i]));
+                }
+                return toggles;
+            };
+            HeaderSlideToggle.softDismissDuration = 250;
+            return HeaderSlideToggle;
+        }());
+        exports.HeaderSlideToggle = HeaderSlideToggle;
+        ;
+    });
+    define("modal", ["require", "exports"], function (require, exports) {
+        "use strict";
+        Object.defineProperty(exports, "__esModule", { value: true });
+        var Modal = (function () {
+            function Modal(element) {
+                this.element = element;
+                this.isPositionedAbsolute = this.element.classList.contains("is-with-backdrop");
+                this.isDanger = this.element.classList.contains("is-danger");
+            }
+            Modal.prototype.toggle = function () {
+                if (!this.isPositionedAbsolute) {
+                    this.element.style.top = document.scrollingElement.scrollTop + "px";
+                }
+                this.element.classList.toggle("is-active");
+            };
+            Modal.find = function (query) {
+                var elements = document.querySelectorAll(query);
+                var toggles = [];
+                for (var i = 0; i < elements.length; i++) {
+                    toggles.push(new Modal(elements[i]));
+                }
+                return toggles;
+            };
+            return Modal;
+        }());
+        exports.Modal = Modal;
+        ;
+        var ModalToggle = (function () {
+            function ModalToggle(element) {
+                var _this = this;
+                this.element = element;
+                this.modal = new Modal(document.querySelector(element.getAttribute("data-modal")));
+                element.addEventListener("click", function () { _this.modal.toggle(); });
+            }
+            ModalToggle.find = function (query) {
+                var elements = document.querySelectorAll(query);
+                var toggles = [];
+                for (var i = 0; i < elements.length; i++) {
```

Comment:
You can reduce your code and make it more elegant by using a `forEach` here and other places. It's more maintainable